scripts/update_ziggurat_tables

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env Rscript
+zig_cpp_tables <- function(template, dest) {
+  header <- "// Generated by scripts/update_ziggurat_tables - do not edit"
+
+  n <- 256
+  dat <- zig_constants(n)
+  r <- dat$r
+  v <- dat$v
+  x <- c(v / f(r), intervals(n, r, v))
+  y <- x[-1] / x[seq_len(n)]
+
+  format <- function(z) {
+    s <- vapply(z, deparse, "", control = "digits17")
+    paste(
+      strwrap(paste(s, collapse = ", "), width = 70, indent = 4, exdent = 4),
+      collapse = "\n")
+  }
+
+  txt <- glue_whisker(template, list(x = format(x), y = format(y)))
+  writeLines(txt, dest)
+}
+
+
+f <- function(x) {
+  exp(-x^2 / 2)
+}
+
+
+f_inv <- function(y) {
+  sqrt(-2 * log(y))
+}
+
+
+f_int <- function(r) {
+  pnorm(r, lower.tail = FALSE) / dnorm(0)
+}
+
+
+intervals <- function(n, r, v) {
+  x <- numeric(n)
+  for (i in seq_len(n - 1)) {
+    x[i] <- if (i == 1) r else f_inv(f(x[i - 1]) + v / x[i - 1])
+  }
+  x
+}
+
+
+zig_constants <- function(n, tolerance = 1e-10) {
```

Comment:
Enough precision here too?

Reply:
I think so - this is 2 orders more than we get than by default. To do this "properly" we'd want to use long doubles really. I've seen implementations with these numbers only ok to 5 digits though, and ours is more accurate than Doornik (only differs in the ~6th place I think)